code/game/objects/items/devices/motion_detector.dm

```diff
@@ -92,32 +96,42 @@
 	playsound(loc, 'sound/items/detector.ogg', 60, 0, 7, 2)
 
 	var/detected
+	var/status = MOTION_DETECTOR_HOSTILE
 	for(var/mob/living/M in orange(detector_range, human_user))
 		if(!isturf(M.loc))
 			continue
-		if(world.time > M.l_move_time + 20)
-			continue //hasn't moved recently
 		if(isrobot(M))
 			continue
 		if(ishuman(M))
 			var/mob/living/carbon/human/H = M
-			if(H.get_target_lock(iff_signal))
-				continue //device checks for IFF data; if it matches, skip.
+			if(H.get_target_lock(iff_signal)) //device checks for IFF data and status
```

Comment:
To avoid spam with dead and cold bodies you'll might want to filter some things:
Monkeys (return)
Alive allies (green blip)
Dead warm bodies (the dead blip)
Dead cold bodies (return)

Reply:
Probably outdated now.

I think keeping monkey detection in is a good idea in balance as marines will want to kill these to deny larva, and if any remain by the time they're planetside, xenos deserve to lose them.

Unrevivable corpse finding is more situational, but you may still have cause to locate them in order to salvage supplies, therefore I've made pinging for living allies, and corpses, both revivable and unrevivable, to be toggleable.